src/checkqueue.h

```diff
@@ -180,9 +186,7 @@ class CCheckQueueControl
     {
         // passed queue is supposed to be unused, or NULL
         if (pqueue != NULL) {
-            assert(pqueue->nTotal == pqueue->nIdle);
-            assert(pqueue->nTodo == 0);
-            assert(pqueue->fAllOk == true);
+            assert(pqueue->IsIdle());
```

Comment:
Good point -- fixed so that the function gets called outside the assert (commit squashed).